foghorn_python/scripts/vcf_comp_funcs.py

```diff
@@ -0,0 +1,138 @@
+		return sample_indexMS, lines_holderMS
+	elif GT in {".|.","./."}:
+                string = "".join(["Sample# ", str(sample_index), " has a both alleles missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	elif GT in {".|2", "2|.", "2/.", "./2", ".|1", "1|.", "1/.", "./1",".|0", "0|.", "0/.", "./0"}:
+		string = "".join(["Sample# ", str(sample_index), " has asingle allele missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	else:
+		return None
+
+
+def GT(line, out, mis_vars):
+        if not line.startswith("##") and not line.startswith("#"):  # Need to spend time making robust and reporting file format and missingness errors
+                fields = line.strip().split()
+		if "GT" not in fields[8]:
+			sys.exit("No genotype tag in format field")# raise error that there is no GT fields in 
+                sample_index = sample_indexMS = 0
+                lines_holderMS = [] # refresh every iteration need to check if this has any effect on the memory management of large datasets
+		lines_holder = []
+		mis = []
+		lines_holder.extend(fields[:4]) # append first few columns 
+		if "," in fields[4]:
+			lines_holderMS.extend(fields[:4])
+			lines_holder.append(str(str(fields[4]).split(",")[0])) #ALT1
+               		lines_holderMS.append(str(str(fields[4]).split(",")[1])) # ALT2
+               		for column in fields[9:]:
+   	        		GT = column.strip().split(":")[0]
+                      		GT_comp(GT, lines_holder, sample_index, mis, fields)
+                     		sample_index = sample_index + 1
+               			mult_al(GT, lines_holderMS, sample_indexMS, mis)
+                		sample_indexMS = sample_indexMS + 1
+		else:
+			lines_holder.append(fields[4])
+               		for column in fields[9:]:
+               	 		GT = column.strip().split(":")[0]
+               	      		GT_comp(GT, lines_holder, sample_index, mis, fields)
+               	        	sample_index = sample_index + 1
+           	lines_holder.append("")
+               	lines_holderMS.append("")
+		mis.append("")
+                out.write('\t'.join(map(str,lines_holder)))
+                out.write("\n")
+		mis_vars.write('\t'.join(map(str,mis)))
+                if len(lines_holderMS) > 1: # the "" appendage makes the list non-empty need to convert to hexdecimal again to compare to foghorn 
+                        out.write('\t'.join(map(str,lines_holderMS)))
+			mis_vars.write('\t'.join(map(str,mis)))
+                        out.write('\n')
+        elif not line.startswith("##") and line.startswith("#"):
+                field = line.strip().split()
+                out.write("#CHR"+ '\t' + "POS" + '\t' + "Ref" + '\t' + "ALT"+ '\t'.join(map(str,field[9:])))
+                out.write('\n')
+
+def DS(line,out):
```

Comment:
It looks like this method does the same thing as GT method above except on different values. Could you try combine these two methods?